wgengine/router/router_linux.go

```diff
@@ -1395,3 +1405,40 @@ func nlAddrOfPrefix(p netip.Prefix) *netlink.Addr {
 		IPNet: netipx.PrefixIPNet(p),
 	}
 }
+
+// Function to check if a file contains a specific string
+func fileContainsString(filePath, searchString string) (bool, error) {
+	data, err := os.ReadFile(filePath)
+	if err != nil {
+		return false, err
+	}
+
+	return strings.Contains(string(data), searchString), nil
+}
+
+// Function to determine if the current device is UDM Pro
+func isUDMPro() bool {
```

Comment:
Please put isUDMPro() in `version/distro` instead of here, and define a new `Distro("udmpro")` to check here.

`if distro.Get() == distro.UDMPro`